TailLight/vfeature.cpp

```diff
@@ -199,37 +202,38 @@ Routine Description:
         return STATUS_BUFFER_TOO_SMALL;
     }
 
-    TailLightReport* packet = nullptr;
-    NTSTATUS status = WdfRequestRetrieveInputBuffer(Request, sizeof(TailLightReport), (void**)&packet, NULL);
-    if (!NT_SUCCESS(status) || !packet) {
-        KdPrint(("TailLight: WdfRequestRetrieveInputBuffer failed 0x%x, packet=0x%p\n", status, packet));
+    status = WdfRequestRetrieveInputBuffer(Request, sizeof(TailLightReport), (void**)&report, NULL);
```

Comment:
Why are you provided a pointer to a self-allocated buffer (`report`) when calling `WdfRequestRetrieveInputBuffer`?

My understanding is that `WdfRequestRetrieveInputBuffer` will give you a direct pointer to  the existing buffer associated with the input request. This is at least how the existing code functions. It's then no point in allocating a buffer in advance with `ExAllocatePool2` as you're doing here.